src/components/inputs/Selector-WorkflowState.tsx

```diff
@@ -29,9 +29,30 @@ export const WorkflowStateSelector = ({
     }
   }
 
+  const handleKeyDown = useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>) => {
+      event.stopPropagation()
+      if (event.key === 'Escape') {
+        setAnchorEl(null)
+      }
+    },
+    [setAnchorEl],
+  )
+
```

Comment:
We are not passing `handleKeyDown` as a prop from parent component to child that relies on referential equality to avoid unnecessary re-renders. So I think `useCallback` here is not necessary. Is there any benefit we get with this or we are using it for some specific reason?